contracts/scripts/deployRound.ts

```diff
@@ -0,0 +1,143 @@
+import { Keypair } from 'maci-domainobjs'
+
+import { deployMaciFactory } from '../utils/deployment'
+import { MaciParameters } from '../utils/maci'
+
+async function main() {
+  const [deployer] = await ethers.getSigners()
+  let maciFactory = await deployMaciFactory(deployer)
+  await maciFactory.deployTransaction.wait()
+
+  const FundingRoundFactory = await ethers.getContractFactory(
+    'FundingRoundFactory',
+    deployer
+  )
+  const fundingRoundFactory = await FundingRoundFactory.deploy(
+    maciFactory.address
+  )
+  await fundingRoundFactory.deployTransaction.wait()
+
+  const transferOwnershipTx = await maciFactory.transferOwnership(
+    fundingRoundFactory.address
+  )
+  await transferOwnershipTx.wait()
+
+  const userRegistryType = process.env.USER_REGISTRY_TYPE || 'simple'
+  let userRegistry: Contract
+  if (userRegistryType === 'simple') {
+    const SimpleUserRegistry = await ethers.getContractFactory(
+      'SimpleUserRegistry',
+      deployer
+    )
+    userRegistry = await SimpleUserRegistry.deploy()
+    await userRegistry.deployTransaction.wait()
+  } else if (userRegistryType === 'brightid') {
+    const BrightIdUserRegistry = await ethers.getContractFactory(
+      'BrightIdUserRegistry',
+      deployer
+    )
+    // TODO sort out if these arguments are sensible
+    userRegistry = await BrightIdUserRegistry.deploy(
+      '0x636c722e66756e64000000000000000000000000000000000000000000000000', // Do we have a test `context`?
+      fundingRoundFactory.address // Not sure if `verifier` should be EOA or contract account
+    )
+    await userRegistry.deployTransaction.wait()
+  } else {
+    throw new Error('unsupported user registry type')
+  }
+  const setUserRegistryTx = await fundingRoundFactory.setUserRegistry(
+    userRegistry.address
+  )
+  await setUserRegistryTx.wait()
+
+  const recipientRegistryType = process.env.RECIPIENT_REGISTRY_TYPE || 'simple'
+  let recipientRegistry: Contract
+  if (recipientRegistryType === 'simple') {
+    const SimpleRecipientRegistry = await ethers.getContractFactory(
+      'SimpleRecipientRegistry',
+      deployer
+    )
+    recipientRegistry = await SimpleRecipientRegistry.deploy(
+      fundingRoundFactory.address
+    )
+    await recipientRegistry.deployTransaction.wait()
+  } else if (recipientRegistryType === 'optimistic') {
+    const OptimisticRecipientRegistry = await ethers.getContractFactory(
+      'OptimisticRecipientRegistry',
+      deployer
+    )
+    recipientRegistry = await OptimisticRecipientRegistry.deploy(
+      ethers.BigNumber.from(10).pow(ethers.BigNumber.from(18)).div(10),
+      300,
+      fundingRoundFactory.address
+    )
+    await recipientRegistry.deployTransaction.wait()
+  } else {
+    throw new Error('unsupported recipient registry type')
+  }
+
+  const setRecipientRegistryTx = await fundingRoundFactory.setRecipientRegistry(
+    recipientRegistry.address
+  )
+  await setRecipientRegistryTx.wait()
+
```

Comment:
So my primary question here - everything above this line looks nearly identical to the deploy.ts script. Why create a new script? Why create a situation where we need to maintain the same code in 2 places?

Reply:
One benefit again of refactoring these to Hardhat tasks is that tasks can call other tasks, which would allow us to modularize these scripts.

Reply:
Mainly did it for simplicity of not needing to fetch the FundingRoundFactory address. No good reason outside of that, just wanted a clean single script for now since if you look at the deployTestRound we hardcoded this factory address. I am happy to take this part out and just create a deployRinkebyRound part, just wanted to reduce the amount of manual intervention as possible.

Reply:
I'm ok with this approach for now. We could save refactor to tasks later.

Reply:
Here's an example of them in action for reference: https://github.com/darkforest-eth/eth/blob/master/tasks/deploy.ts

Reply:
Sounds good, i'll read up on some of the hardhat task stuff over the weekend to see how this can be simplified as much as possible. I do agree that it would be much nicer to remove this duplication, and hopefully there may be a way to just pass some of this info along/use environment variables to customize this using tasks.